### PR TITLE
Add documentation for migrating from TAS to k8s

### PR DESCRIPTION
**Pull request by ssapra**
[#183906107](https://www.pivotaltracker.com/story/show/183906107)

Authored-by: Shaan Sapra <shsapra@vmware.com>

Name the branches to merge this change with or enter "None".

### CODE REVIEW

File: migrate-mysql.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+This topic describes how to migrate data from VMware Tanzu SQL with MySQL for VMs to VMware MySQL for Kubernetes.
+
+Tanzu SQL for VMs only supports MySQL 5.7 while MySQL for Kubernetes only supports MySQL 8.
+
+Before upgrading using a logical dump, please 
+
+## <a id='prerequisites'></a>  Prerequisites
+
+Before you can back up MySQL service instances and restore to MySQL for Kubernetes, you must have:
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:14]: drop "have" (or match the tense below, eg "you must have... **referred** to the documentation", but I think present tense is better)

File: migrate-mysql.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,127 @@
+---
+title: Migrating from TAS to Kubernetes
+owner: MySQL
+---
+
+This topic describes how to migrate data from VMware Tanzu SQL with MySQL for VMs to VMware MySQL for Kubernetes.
+
+Tanzu SQL for VMs only supports MySQL 5.7 while MySQL for Kubernetes only supports MySQL 8.
+
+Before upgrading using a logical dump, please
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:10]: I think this is the section below.

File: migrate-mysql.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+Tanzu SQL for VMs only supports MySQL 5.7 while MySQL for Kubernetes only supports MySQL 8.
+
+Before upgrading using a logical dump, please 
+
+## <a id='prerequisites'></a>  Prerequisites
+
+Before you can back up MySQL service instances and restore to MySQL for Kubernetes, you must have:
+
+* Refer to the [documentation](https://dev.mysql.com/doc/refman/8.0/en/mysql-nutshell.html) for all the changes in MySQL 8.
+* Take note of which schemas you wish to migrate. The default schema for every instance is `service_instance_db`.
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:17]: "The default schema for MySQL instances on TAS is `service_instance_db`" (or so)

File: migrate-mysql.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+* Have a local machine with the following installed:
+  * `mysqldump` 5.7 to connect to MySQL service instances externally, take a logical backup, and store the backup
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:19]: Is this right? Maybe I'm thinking of Postgres which wants you to dump with the newer major version to do an upgrade.

File: migrate-mysql.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  * `mysql` 8 to connect to MySQL on Kubernetes externally and load the backup
+  * The Cloud Foundry Command-Line Interface (cf CLI). See [Installing the cf CLI](https://docs.pivotal.io/application-service/2-10/cf-cli/install-go-cli.html). 
+
+## <a id='backing-up'></a>  Backing up with mysqldump
+
+For each MySQL for VMs service instance, follow the steps from [this documentation](https://docs.vmware.com/en/VMware-Tanzu-SQL-with-MySQL-for-VMs/2.10/tanzu-mysql-vms/GUID-backup-mysqldump.html#back-up-and-restore-a-tanzu-sql-for-vms-logical-backup-1) how to manually create a logical backup with `mysqldump`.
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:25]: ```follow the [steps to manually create a logical backup with `mysqldump`](link)```

(it's an old HTML principle: don't say "here" or "this link". I think it was Tim Berners Lee's rule.)

File: migrate-mysql.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+FLUSH PRIVILEGES;
+```
+
+See [MySQL for Kubernetes documentation](https://docs.vmware.com/en/VMware-Tanzu-SQL-with-MySQL-for-Kubernetes/1.6/tanzu-mysql-k8s/GUID-connecting-apps.html#create-a-database-and-privileged-mysql-user-for-the-app) for more information on creating a user for the associated database.
+
+4. Ensure the MySQL pod can be access external and connect with the newly created user to load the backup file
+
+> Note: Make sure to use a mysql 8.0 client from your workstation when connecting to the MySQL service
+
+See [Access the MySQL Server from an External IP Address](https://docs.vmware.com/en/VMware-Tanzu-SQL-with-MySQL-for-Kubernetes/1.6/tanzu-mysql-k8s/GUID-accessing.html#access-the-mysql-server-from-an-external-ip-address) to configure the MySQL instance for external access.
```
Comment (dsharp-pivotal) [inline migrate-mysql.html.md.erb:107]: Exposing the port externally might not be desirable.

kubectl port-forward is also an option. However it does mean shoving the entire database through the api-server, so an external IP might be faster and kinder to the api-server.

I was also thinking of ways to load the dump file from within the k8s cluster.
- One could use kubectl cp to move the file to a Pod that can access the mysql Service. That would also shove the file through the api-server, though.
- Store the dump on s3/etc and use a pod to retrieve and load it.

Maybe more creative solutions should be left as an exercise for the reader.

Comment (ssapra) [reply migrate-mysql.html.md.erb:107]: Agreed. I do mention at the end to switch back to ClusterIP but users may miss that or may not be interested in temporarily exposing the MySQL server.

The trouble with using some other non-MySQL Pod is that it needs the mysql 8 client.

Perhaps I can provide a list of options and leave it up the user to pick the appropriate one.

**Pull request merged**